klauspost/minio
# Usage: Add partial bucket updates

## Description

Add partial bucket updates.

PR against https://github.com/minio/minio/pull/12229

## Motivation and Context

Usage is mainly used for metrics and with big buckets scanning can be quite time consuming.

This means that there can be a long while between updates or one bucket can hold up everything.

This change collects stats as they are seen and forwards them upstream.

## How to test this PR?

Clean the usage cache and observe as scanning populates the update cache.

Also observe that usage remains stable while scanning non-updated folders.

## Types of changes
